src/transformers/models/bloom/modeling_bloom.py

```diff
@@ -318,7 +301,19 @@ def forward(self, input, mask, max_positions, prefix_mask):
             .view(1, 1, max_positions, max_positions)
             .to(input.device)
         )
-        mask_output, padded_causal_mask = self.mask_func(input, mask, causal_mask, prefix_mask)
+
+        if prefix_length is not None:
+            prefix_mask = torch.zeros(
+                (prefix_length.shape[-1], 1, max_positions, max_positions),
+                dtype=torch.bool
+            )
+            for idx in range(prefix_length.shape[-1]):
+                prefix_mask_length = int(prefix_length[idx])
+                prefix_mask[idx, :, :prefix_mask_length, :prefix_mask_length] = 1
+            prefix_mask = prefix_mask.to(input.device)
+            causal_mask = (causal_mask+prefix_mask).bool()
+
+        mask_output, padded_causal_mask = self.mask_func(input, mask, causal_mask)
```

Comment:
I'll need to check this, but why is it `prefix_length.shape[-1]` instead of `prefix_length.shape[0]`, shouldn't the batch dimension be the first dimension?

Reply:
You're right. Let me change this. This should be in fact `prefix_length.shape[0]`.